grammars/type/grammar.js

```diff
@@ -0,0 +1,16 @@
+/// <reference types="tree-sitter-cli/dsl" />
+// @ts-check
+
+module.exports = grammar(require('../ocaml/grammar'), {
+  name: 'ocaml_type',
+
+  rules: {
+    type: $ => $._type
+  }
+})
+
+// Make 'type' the first rule
```

Comment:
Why do we need this extra magic?

Reply:
Without this, we'd have to reuse the `compilation_unit` entrypoint of the ocaml grammar, so `'a` would be parsed as
```scheme
(compilation_unit (type_variable))
```
instead of
```scheme
(type (type_variable))
```

The tree-sitter extension mechanism is not really made for changing the entrypoint of a grammar, but for overriding some rules to create a variation of the same grammar.